File: app/src/main/java/com/ergv/glScreenSavers/TableOfContents.java
package com.ergv.glScreenSavers;

import android.app.Activity;
import android.app.ListActivity;
import android.content.Intent;
import android.os.Bundle;
import android.util.SparseArray;
import android.view.View;
import android.widget.AdapterView;
import android.widget.AdapterView.OnItemClickListener;
import android.widget.SimpleAdapter;

import com.ergv.glScreenSavers.JuliaSet.JuliaSetActivity;
import com.ergv.glScreenSavers.Polyhedrons.PolyhedronsActivity;
import com.ergv.glScreenSavers.bricksShader.BricksShaderActivity;
import com.ergv.glScreenSavers.quantizedColor.QuantizedColorActivity;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class TableOfContents extends ListActivity {
    private static final String ITEM_IMAGE = "item_image";
    private static final String ITEM_TITLE = "item_title";
    private static final String ITEM_SUBTITLE = "item_subtitle";

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setTitle(R.string.toc);
        setContentView(R.layout.table_of_contents);

        // Initialize data
        final List<Map<String, Object>> data = new ArrayList<Map<String, Object>>();
        final SparseArray<Class<? extends Activity>> activityMapping = new SparseArray<Class<? extends Activity>>();

        int i = 0;
        {
            final Map<String, Object> item = new HashMap<String, Object>();
            item.put(ITEM_IMAGE, R.drawable.bricks_icon);
            item.put(ITEM_TITLE, getText(R.string.spinning_polyhedrons));
            item.put(ITEM_SUBTITLE, getText(R.string.spinning_polyhedrons_subtitle));
            data.add(item);
            activityMapping.put(i++, PolyhedronsActivity.class);
        }
        {
            final Map<String, Object> item = new HashMap<String, Object>();
            item.put(ITEM_IMAGE, R.drawable.ic_lesson_one);
            item.put(ITEM_TITLE, getText(R.string.quantized_color));
            item.put(ITEM_SUBTITLE, getText(R.string.quantized_color_subtitle));
            data.add(item);
            activityMapping.put(i++, QuantizedColorActivity.class);
        }
        {
            final Map<String, Object> item = new HashMap<String, Object>();
            item.put(ITEM_IMAGE, R.drawable.bricks_icon);
            item.put(ITEM_TITLE, getText(R.string.julia_set));
            item.put(ITEM_SUBTITLE, getText(R.string.julia_set_subtitle));
            data.add(item);
            activityMapping.put(i++, JuliaSetActivity.class);
        }
        {
            final Map<String, Object> item = new HashMap<String, Object>();
            item.put(ITEM_IMAGE, R.drawable.bricks_icon);
            item.put(ITEM_TITLE, getText(R.string.bricks_shader));
            item.put(ITEM_SUBTITLE, getText(R.string.bricks_shader_subtitle));
            data.add(item);
            activityMapping.put(i++, BricksShaderActivity.class);
        }




        final SimpleAdapter dataAdapter = new SimpleAdapter(this, data, R.layout.toc_item, new String[]{ITEM_IMAGE, ITEM_TITLE, ITEM_SUBTITLE}, new int[]{R.id.Image, R.id.Title, R.id.SubTitle});
        setListAdapter(dataAdapter);

        getListView().setOnItemClickListener(new OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view,
                                    int position, long id) {
                final Class<? extends Activity> activityToLaunch = activityMapping.get(position);

                if (activityToLaunch != null) {
                    final Intent launchIntent = new Intent(TableOfContents.this, activityToLaunch);
                    startActivity(launchIntent);
                }
            }
        });
    }
}

File: app/src/main/java/com/ergv/glScreenSavers/TableOfContents.kt
package com.ergv.glScreenSavers

import android.app.Activity
import android.app.ListActivity
import android.content.Intent
import android.os.Bundle
import android.util.SparseArray
import android.widget.AdapterView.OnItemClickListener
import android.widget.SimpleAdapter
import com.ergv.glScreenSavers.MandelbrotSet.MandelbrotSetActivity
import com.ergv.glScreenSavers.Polyhedrons.PolyhedronsActivity
import com.ergv.glScreenSavers.bricksShader.BricksShaderActivity
import com.ergv.glScreenSavers.geodeticBodies.GeodeticBodiesActivity
import java.util.*


class TableOfContents : ListActivity() {
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setTitle(R.string.toc)
        setContentView(R.layout.table_of_contents)

        // Initialize data
        val data: MutableList<Map<String, Any?>> = ArrayList()
        val activityMapping = SparseArray<Class<out Activity?>>()
        var i = 0
        run {
            val item: MutableMap<String, Any?> = HashMap()
            item[ITEM_IMAGE] = R.drawable.bricks_icon
            item[ITEM_TITLE] = getText(R.string.geodetic_bodies_title)
            item[ITEM_SUBTITLE] = getText(R.string.geodetic_bodies_subtitle)
            data.add(item)
            activityMapping.put(i++, GeodeticBodiesActivity::class.java)
        }
        run {
            val item: MutableMap<String, Any?> = HashMap()
            item[ITEM_IMAGE] = R.drawable.bricks_icon
            item[ITEM_TITLE] = getText(R.string.spinning_polyhedrons)
            item[ITEM_SUBTITLE] = getText(R.string.spinning_polyhedrons_subtitle)
            data.add(item)
            activityMapping.put(i++, PolyhedronsActivity::class.java)
        }

        run {
            val item: MutableMap<String, Any?> = HashMap()
            item[ITEM_IMAGE] = R.drawable.bricks_icon
            item[ITEM_TITLE] = getText(R.string.mandelbrot_set)
            item[ITEM_SUBTITLE] = getText(R.string.mandelbrot_set_subtitle)
            data.add(item)
            activityMapping.put(i++, MandelbrotSetActivity::class.java)
        }
        run {
            val item: MutableMap<String, Any?> = HashMap()
            item[ITEM_IMAGE] = R.drawable.bricks_icon
            item[ITEM_TITLE] = getText(R.string.bricks_shader)
            item[ITEM_SUBTITLE] = getText(R.string.bricks_shader_subtitle)
            data.add(item)
            activityMapping.put(i++, BricksShaderActivity::class.java)
        }
        val dataAdapter = SimpleAdapter(this, data, R.layout.toc_item, arrayOf(ITEM_IMAGE, ITEM_TITLE, ITEM_SUBTITLE), intArrayOf(R.id.Image, R.id.Title, R.id.SubTitle))
        listAdapter = dataAdapter
        listView.onItemClickListener = OnItemClickListener { parent, view, position, id ->
            val activityToLaunch = activityMapping[position]
            if (activityToLaunch != null) {
                val launchIntent = Intent(this@TableOfContents, activityToLaunch)
                startActivity(launchIntent)
            }
        }
    }

    companion object {
        private const val ITEM_IMAGE = "item_image"
        private const val ITEM_TITLE = "item_title"
        private const val ITEM_SUBTITLE = "item_subtitle"
        private const val ITEM_ID = "item_id"
    }
}